python/paddle/fluid/contrib/slim/tests/test_dygraph_qat.py

```diff
@@ -0,0 +1,181 @@
+    def test_qat(self):
+        dyquant_pass = DygraphQuantizationPass()
+        dyquant_pass.prepare_quantize()
+        with fluid.dygraph.guard():
+            mnist = MNIST()
+            dyquant_pass.quantize_qat(mnist)
+            adam = AdamOptimizer(
+                learning_rate=0.001, parameter_list=mnist.parameters())
+            train_reader = paddle.batch(
+                paddle.dataset.mnist.train(), batch_size=32, drop_last=True)
+            test_reader = paddle.batch(
+                paddle.dataset.mnist.test(), batch_size=32)
+
+            epoch_num = 2
+            for epoch in range(epoch_num):
+                mnist.train()
+                for batch_id, data in enumerate(train_reader()):
+                    x_data = np.array([x[0].reshape(1, 28, 28)
+                                       for x in data]).astype('float32')
+                    y_data = np.array(
+                        [x[1] for x in data]).astype('int64').reshape(-1, 1)
+
+                    img = fluid.dygraph.to_variable(x_data)
+                    label = fluid.dygraph.to_variable(y_data)
+
+                    out = mnist(img)
+                    acc = fluid.layers.accuracy(out, label)
+                    loss = fluid.layers.cross_entropy(out, label)
+                    avg_loss = fluid.layers.mean(loss)
+                    avg_loss.backward()
+                    adam.minimize(avg_loss)
+                    mnist.clear_gradients()
+                    if batch_id % 100 == 0:
+                        print(
+                            "Train | At epoch {} step {}: loss = {:}, acc= {:}".
+                            format(epoch, batch_id,
+                                   avg_loss.numpy(), acc.numpy()))
+
+                mnist.eval()
+                for batch_id, data in enumerate(test_reader()):
+                    x_data = np.array([x[0].reshape(1, 28, 28)
+                                       for x in data]).astype('float32')
+                    y_data = np.array(
+                        [x[1] for x in data]).astype('int64').reshape(-1, 1)
+
+                    img = fluid.dygraph.to_variable(x_data)
+                    label = fluid.dygraph.to_variable(y_data)
+
+                    out = mnist(img)
+                    acc_top1 = fluid.layers.accuracy(
+                        input=out, label=label, k=1)
+                    acc_top5 = fluid.layers.accuracy(
+                        input=out, label=label, k=5)
+
+                    if batch_id % 100 == 0:
+                        print(
+                            "Test | At epoch {} step {}: acc1 = {:}, acc5 = {:}".
+                            format(epoch, batch_id,
+                                   acc_top1.numpy(), acc_top5.numpy()))
+            # save weights
+            model_dict = mnist.state_dict()
+            fluid.save_dygraph(model_dict, "save_temp")
+
+        # save inference quantized model
+        dyquant_pass.save_infer_quant_model(
```

Comment:
Maybe not be done in this PR: 

Do we have a test plan for loading the saved model to ensure it can work? Because I just saw save here, it proves it can run but no numerical result is showed.

Reply:
Done.